drivers/sensor/ti/tmp11x/tmp11x.c

```diff
@@ -424,18 +504,68 @@ static int tmp11x_init(const struct device *dev)
 
 	rc = tmp11x_write_config(dev, TMP11X_CFGR_AVG, cfg->oversampling);
 
+	rc = tmp11x_write_config(dev, TMP11X_CFGR_ALERT_PIN_POL, cfg->alert_pin_polarity);
+
+	rc = tmp11x_write_config(dev, TMP11X_CFGR_ALERT_MODE, cfg->alert_mode);
+
+	rc = tmp11x_write_config(dev, TMP11X_CFGR_ALERT_DR_SEL, cfg->alert_dr_sel);
+
+#ifdef CONFIG_TMP11X_TRIGGER
+	drv_data->dev = dev;
+	rc = tmp11x_init_interrupt(dev);
+	if (rc < 0) {
+		LOG_ERR("%s: Failed to initialize alert pin", dev->name);
+		return rc;
+	}
+#endif /* CONFIG_TMP11X_TRIGGER */
+
 	return rc;
 }
 
-#define DEFINE_TMP11X(_num) \
-	static struct tmp11x_data tmp11x_data_##_num; \
-	static const struct tmp11x_dev_config tmp11x_config_##_num = { \
-		.bus = I2C_DT_SPEC_INST_GET(_num), \
-		.odr = DT_INST_PROP(_num, odr), \
-		.oversampling = DT_INST_PROP(_num, oversampling), \
-	}; \
-	SENSOR_DEVICE_DT_INST_DEFINE(_num, tmp11x_init, NULL, \
-		&tmp11x_data_##_num, &tmp11x_config_##_num, POST_KERNEL, \
-		CONFIG_SENSOR_INIT_PRIORITY, &tmp11x_driver_api);
+#ifdef CONFIG_PM_DEVICE
+static int tmp11x_pm_control(const struct device *dev, enum pm_device_action action)
+{
+	int ret = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		const struct tmp11x_dev_config *cfg = dev->config;
+
+		ret = tmp11x_write_config(dev, TMP11X_CFGR_CONV, cfg->odr);
+		if (ret < 0) {
+			LOG_ERR("Failed to resume TMP11X");
+		}
+		break;
+
+	case PM_DEVICE_ACTION_SUSPEND:
+		ret = tmp11x_write_config(dev, TMP11X_CFGR_MODE, TMP11X_MODE_SHUTDOWN);
+		if (ret < 0) {
+			LOG_ERR("Failed to suspend TMP11X");
+		}
+		break;
+
```

Comment:
You  may want to add an assert to ensure sensor is not placed on a power domain. If the power domain is suspended, it would clear the sensor's registers, which the driver is not built to handle currently (missing TURN_OFF action) :)
```
BUILD_ASSERT(!DT_INST_NODE_HAS_PROP(_num, power_domains), "Driver does not support power domain");
```